### PR TITLE
ADD sass-lint sassdoc dependencies

### PR DESCRIPTION
**Pull request by occupant**
- adds a gulp task for sass-lint `lint`, which defines and enforces specific css/sass formatting and ordering
- adds a gulp task for sassdoc `sassdoc`, which scrapes sass source files to generate documentation from the comments
- adds DS_Store files to gitignore (OSX is messy)
- reformats some of our sass comments to adhere with our lint rules and sassdoc requirements

### CODE REVIEW

**Comment by occupant** [comment]
I'm thinking a squash and merge policy would be a good idea. Just to keep the repo clean.

File: config/.sass-lint.yml
Comment (richardtape) [inline config/.sass-lint.yml:51]: `bem-depth: 2` and then `max-depth: 1`? Probably I'm misreading or misunderstanding but that just looks weird, so thought I'd ask.

File: src/sass/base/_typography.scss
Comment (richardtape) [inline src/sass/base/_typography.scss]: Missing "for"
...typographic stiles _for_ copy text.

**Review by richardtape — CHANGES_REQUESTED** [review]
Might as well test our "Request Changes" strategy, even if I am being captain pedantic.

File: src/sass/base/_typography.scss
Comment (occupant) [reply src/sass/base/_typography.scss]: Absolutely :)

File: config/.sass-lint.yml
Comment (occupant) [reply config/.sass-lint.yml:51]: I don't recall - I'll test and check. It does look strange....

Comment (occupant) [reply config/.sass-lint.yml:51]: Ok, I tested and what it's doing is this.

`bem-depth: 2` refers to the setting - 0 is ignore, 1 is warning, 2 is error (preventing the compile)
`max-depth: 1` refers to the number of elements allowed in a BEM selector (the E in BEM). 

I think these are a good place to start. The whole point of BEM is to prevent selector depth, so only allowing a single element seems reasonable to me.

Comment (richardtape) [reply config/.sass-lint.yml:51]: Thanks for the clarification. Makes sense!

**Review by richardtape — APPROVED** [review]

**Comment by richardtape** [comment]
I'll wait for @hazuremon to finish his review before merging. Do you think that strategy works going ahead?

**Comment by occupant** [comment]
Generally I think it's good. It might be a bit of micromanagment in the short term, but good long term.

We probably only need one approver, but this is a nice test case.

**Review by hazuremon — APPROVED** [review]
Looks good to me as well. Thanks James!

**Comment by occupant** [comment]
Cool, I'll try a squash and merge so we can condense it all to one commit. Thanks both!

**Pull request merged**
